iina/PlayerCore.swift

```diff
@@ -579,17 +580,19 @@ class PlayerCore: NSObject {
     info.disableOSDForFileLoading = true
     guard let path = mpvController.getString(MPVProperty.path) else { return }
     info.currentURL = URL(fileURLWithPath: path)
-    // add files in same folder
-    if ud.bool(forKey: Preference.Key.playlistAutoAdd) {
-      autoLoadFilesInCurrentFolder()
-    }
-    // auto load matched subtitles
-    if let matchedSubs = info.matchedSubs[path] {
-      for sub in matchedSubs {
-        loadExternalSubFile(sub)
+    backgroundQueue.async {
```

Comment:
So I had a chance to try this algorithm, and it's falling apart in my real-world usage. Basically, I have a video in my downloads folder, which has slightly over 1500 items. If I play it, IINA will idle at 100% CPU as it goes through this method. I can attach a stackshot if you'd like.